orienteer-architect/src/main/java/org/orienteer/architect/component/widget/OArchitectEditorWidget.java

```diff
@@ -0,0 +1,202 @@
+package org.orienteer.architect.component.widget;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import com.google.common.reflect.TypeToken;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.orientechnologies.orient.core.db.document.ODatabaseDocument;
+import com.orientechnologies.orient.core.metadata.schema.OClass;
+import com.orientechnologies.orient.core.metadata.schema.OSchema;
+import com.orientechnologies.orient.core.record.impl.ODocument;
+import de.agilecoders.wicket.webjars.request.resource.WebjarsCssResourceReference;
+import de.agilecoders.wicket.webjars.request.resource.WebjarsJavaScriptResourceReference;
+import org.apache.wicket.Component;
+import org.apache.wicket.WicketRuntimeException;
+import org.apache.wicket.ajax.AbstractDefaultAjaxBehavior;
+import org.apache.wicket.ajax.AjaxRequestTarget;
+import org.apache.wicket.ajax.attributes.CallbackParameter;
+import org.apache.wicket.behavior.Behavior;
+import org.apache.wicket.markup.head.CssReferenceHeaderItem;
+import org.apache.wicket.markup.head.IHeaderResponse;
+import org.apache.wicket.markup.head.JavaScriptHeaderItem;
+import org.apache.wicket.markup.head.OnLoadHeaderItem;
+import org.apache.wicket.markup.html.WebMarkupContainer;
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.model.ResourceModel;
+import org.apache.wicket.request.IRequestParameters;
+import org.apache.wicket.request.cycle.RequestCycle;
+import org.apache.wicket.request.resource.CssResourceReference;
+import org.apache.wicket.request.resource.JavaScriptResourceReference;
+import org.orienteer.architect.OArchitectModule;
+import org.orienteer.architect.util.OArchitectOClass;
+import org.orienteer.architect.util.OArchitectOProperty;
+import org.orienteer.architect.util.OClassJsonDeserializer;
+import org.orienteer.core.component.FAIcon;
+import org.orienteer.core.component.FAIconType;
+import org.orienteer.core.widget.AbstractWidget;
+import org.orienteer.core.widget.Widget;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ru.ydn.wicket.wicketorientdb.security.OSecurityHelper;
+import ru.ydn.wicket.wicketorientdb.security.OrientPermission;
+import ru.ydn.wicket.wicketorientdb.security.RequiredOrientResource;
+import ru.ydn.wicket.wicketorientdb.utils.DBClosure;
+
+import java.lang.reflect.Type;
+import java.util.List;
+
+/**
+ * Editor widget for OrientDB Schema
+ */
+@Widget(id="architect-editor", domain = "document", tab="schemeEditor", selector = OArchitectModule.OARCHITECTOR_CLASS, autoEnable = true)
```

Comment:
Lets think twice about how to name this class for storing diagram. For now it' ok.